frontend/javascripts/oxalis/model/sagas/saga_helpers.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+import type { Action } from "oxalis/model/actions/actions";
+import { type Saga, call, put, select, _takeEvery } from "oxalis/model/sagas/effect-generators";
+import { type Pattern } from "redux-saga";
+import { setBusyBlockingInfoAction } from "oxalis/model/actions/ui_actions";
+
+export function* takeEveryUnlessBusy(
+  actionDescriptor: Pattern,
+  saga: Action => Saga<void>,
+  reason: string,
+): Saga<void> {
+  /*
+   * Similar to _takeEvery, this function can be used to react to
+   * actions to start sagas. However, the difference is that once the given
+   * saga is executed, webKnossos will be marked as busy. When being busy,
+   * following actions which match the actionDescriptor are ignored.
+   * When the given saga finishes, busy is set to false.
+   *
+   * Note that busyBlockingInfo is also used/respected in other places within
+   * webKnossos.
+   */
+
+  function* sagaBusyWrapper(action: Action) {
+    const busyBlockingInfo = yield* select(state => state.uiInformation.busyBlockingInfo);
+    if (busyBlockingInfo.isBusy) {
+      console.warn(
+        `Ignoring ${action.type} request (reason: ${busyBlockingInfo.reason || "null"})`,
+      );
+      return;
+    }
+
+    yield* put(setBusyBlockingInfoAction(true, reason));
+    yield* call(saga, action);
+    yield* put(setBusyBlockingInfoAction(false));
```

Comment:
The busy-info stuff is meant as a sort of mutex. I just checked all occurrences (there are not many) and all use the pattern correctly. As a safe guard, I added an assertion to the corresponding reducer so that an error is thrown in case the mutex is not handled properly (now or in the future). See 24acd467b.